cognite_toolkit/_cdf_tk/warnings.py

```diff
@@ -52,3 +64,40 @@ def __str__(self) -> str:
             for warning in group_list:
                 output.append(f"{'    ' * 2} * {warning!s}")
         return "\n".join(output)
+
+
+@dataclass(frozen=True)
+class GeneralWarning(ToolkitWarning):
+    severity: SeverityLevel | None = None
+    message: str | None = None
```

Comment:
I think maybe this should be a method instead. That will give more flexibility. 
```python
@dataclass(frozen=True)
class GeneralWarning(ToolkitWarning):
    @abstractmethod
    def get_message() -> str:
         raise NotImplementedError()
```

Then
```python
class ToolkitNotSupportedWarning(GeneralWarning):
    functionality: str

   def get_message(self) -> str:
         return f"{self.functionality} is not supported, skipping...
```